clippy_lints/src/iter_without_into_iter.rs

```diff
@@ -104,6 +104,12 @@ fn is_nameable_in_impl_trait(ty: &rustc_hir::Ty<'_>) -> bool {
     !matches!(ty.kind, TyKind::OpaqueDef(..))
 }
 
+fn is_ty_exported(cx: &LateContext<'_>, ty: Ty<'_>) -> bool {
```

Comment:
I think it would be good to document this restriction to public items in the lint description. Maybe under a `### Limitations` section to just under the normal `### What it does?` section.

Reply:
Good idea. In my experience, "Limitations" is under "Why is this bad?". I'll put that there instead, for consistency's sake.

Reply:
Done. What do you think about the wording? Not gonna approve just yet because I'd like to hear your opinion

Reply:
@xFrednet ? :)
(I'm not sure if you saw the comment or if you're just busy)

Reply:
I actually missed your comment, thank you for pointing that out. It's correct, that the `### Limitations` section comes after `### Why is it bad?`